### PR TITLE
Clean up old sessions in the NDK

### PR DESCRIPTION
**Pull request by mrober**
Clean up old sessions in the NDK before initializing and after finalizing a session.

Tested manually with a test app.

### CODE REVIEW

File: firebase-crashlytics-ndk/src/main/java/com/google/firebase/crashlytics/ndk/NdkCrashFilesManager.java [hunk 1]
```diff
@@ -38,10 +43,21 @@ public File getSessionFileDirectory(String sessionId) {
...
 
   @Override
-  public void deleteSessionFilesDirectory(String sessionId) {
+  public void deleteSessionFileDirectory(String sessionId) {
     recursiveDelete(new File(rootPath, sessionId));
   }
 
+  @Override
+  public void cleanOldSessionFileDirectories() {
+    File[] sessionFileDirectories = rootPath.listFiles(File::isDirectory);
```
Comment (kmandrika) [inline firebase-crashlytics-ndk/src/main/java/com/google/firebase/crashlytics/ndk/NdkCrashFilesManager.java:52]: I'd double check with Marc about JDK version requirements - this syntax may not be supported.

Comment (mrichards) [reply firebase-crashlytics-ndk/src/main/java/com/google/firebase/crashlytics/ndk/NdkCrashFilesManager.java:52]: It should work now that we're requiring everyone to update to Java 8. I believe this syntax is 100% resolved in the compiler but it is probably worth testing on an API 19 emulator, which I believe is the oldest we support.

**Review by mrichards — APPROVED** [review]
LGTM, pending confirmation that JDK version isn't an issue.

File: firebase-crashlytics-ndk/src/main/java/com/google/firebase/crashlytics/ndk/NdkCrashFilesManager.java (hunk 1 above)
Comment (mrober) [reply firebase-crashlytics-ndk/src/main/java/com/google/firebase/crashlytics/ndk/NdkCrashFilesManager.java:52]: I tested it on an emulator with API 16 and it works fine. We use this syntax in a few other places already.

**Comment by mrober** [comment]
\test macrobenchmark

**Pull request merged**